h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java

```diff
@@ -21,24 +23,59 @@ public StackedEnsembleMojoModel(String[] columns, String[][] domains, String res
             basePreds = new double[_baseModelNum * _nclasses]; //Proper allocation for multinomial ensemble (class probabilities per base model)
             for(int i = 0; i < _baseModelNum; ++i){
                 for(int j = 0; j < _nclasses; ++j){
-                    basePreds[k] = _baseModels[i].score0(row, basePredsRow)[j+1];
+                    basePreds[k] = _baseModels[i]._mojoModel.score0(_baseModels[i].remapRow(row), basePredsRow)[j + 1];
                     k++;
                 }
             }
         }else if(_nclasses == 2){ //Binomial
             for(int i = 0; i < _baseModelNum; ++i) {
-                _baseModels[i].score0(row, basePredsRow);
+                _baseModels[i]._mojoModel.score0(_baseModels[i].remapRow(row), basePredsRow);
                 basePreds[i] = basePredsRow[2];
             }
         }else{ //Regression
             for(int i = 0; i < _baseModelNum; ++i) { //Regression
-                _baseModels[i].score0(row, basePredsRow);
+                _baseModels[i]._mojoModel.score0(_baseModels[i].remapRow(row), basePredsRow);
                 basePreds[i] = basePredsRow[0];
             }
         }
         _metaLearner.score0(basePreds, preds);
         return preds;
     }
 
+    /**
+     * In stacked ensembles, multiple models may appear. Problem with multiple models present is possibly different
+     * internal order of features. Therefore, the scored row's values must be re-mapped to the internal order of each
+     * model.
+     */
+    static class StackedEnsembleMojoSubModel {
+
+        private final MojoModel _mojoModel;
+        private final int[] _mapping; // Mapping. If null, no mapping is required.
+
+        public StackedEnsembleMojoSubModel(MojoModel mojoModel, int[] mapping) {
+            _mojoModel = mojoModel;
+            _mapping = mapping;
+        }
+
+        /**
+         * Returns a new array represeting row values re-mapped to order given by the underlying submodel.
+         * Order of columns in the row may remain the same, yet a new instance of double[] is returned all the time.
+         *
+         * @param row Row to re-map
+         * @return A new instance of double[] with values re-mapped to order given by the underlying submodel.
+         */
+        public double[] remapRow(final double[] row) {
```

Comment:
Called for every model. Ensembles with many models may share the mapping across multiple models.

Isn't there a (fast enough) way to know if a mapping like this already occurred before and instead of doing the mapping again, just producing a new copy of the once mapped data ?

Reply:
I think hashing might do the job

Reply:
When stored, yes. For a big ensemble of 10+ models, it could save time, as there are practically 2 ways of column ordering right now.

Reply:
I'D like to compare numbers instead of traditional hashes. Much faster. Can be done.

Reply:
I actualy did this in the last commit. We can just delete the last commit if required.